internal/api/v1/application/stage.go

```diff
@@ -262,41 +263,41 @@ func (hc Controller) Staged(c *gin.Context) apierror.APIErrors {
 		return apierror.InternalError(err)
 	}
 
-	cs, err := tekton.NewForConfig(cluster.RestConfig)
+	// Wait for the staging POD to complete, or fail.
```

Comment:
as per my other comment, we should better rely on `Job` states

Reply:
See https://github.com/epinio/epinio/issues/1179 (folded into it, main issue is the block on re-staging when a staging run failed, due mis-detection as still running).